src/compiler/parser.ts

```diff
@@ -5227,18 +5265,38 @@ namespace ts {
 
         // DECLARATIONS
 
-        function parseArrayBindingElement(): ArrayBindingElement {
+        function parseArrayBindingElementInSpeculation(): ArrayBindingElement | Fail {

```

Comment:
As above, I'd recommend `tryParseArrayBindingElement` and `parseArrayBindingElement`.